storage/filesystem/internal/dotgit/dotgit.go

```diff
@@ -103,3 +103,52 @@ func (d *DotGit) Idxfile() (fs.FS, string, error) {
 
 	return nil, "", ErrIdxNotFound
 }
+
+func (d *DotGit) Objectfiles() (fs.FS, []core.Hash, error) {
+	dotGitobjcts := d.fs.Join(d.path, "objects")
+	files, err := d.fs.ReadDir(dotGitobjcts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var objDirs []string
+	reDir, _ := regexp.Compile("[a-z0-9]{2}")
+	for _, f := range files {
+		if f.IsDir() && reDir.MatchString(f.Name()) {
+			objDirs = append(objDirs, f.Name())
+		}
+	}
+
+	var objects []core.Hash
+	reObj, _ := regexp.Compile("[a-z0-9]{38}")
+	for _, dir := range objDirs {
+		objs, err := d.fs.ReadDir(d.fs.Join(dotGitobjcts, dir))
+
+		if err != nil {
+			return nil, nil, err
+		}
+
+		for _, obj := range objs {
+			if reObj.MatchString(obj.Name()) {
+				name := dir + obj.Name()
+				fmt.Println(name)
+				objects = append(objects, core.NewHash(name))
+			}
+		}
+	}
+
+	return d.fs, objects, nil
+}
+
+func (d *DotGit) Objectfile(h core.Hash) (fs.FS, string, error) {
+	hash := h.String()
+	objFile := d.fs.Join(d.path, "objects", hash[0:2], hash[2:40])
+
+	if _, err := d.fs.Stat(objFile); err != nil {
+		if os.IsNotExist(err) {
+			return nil, "", ErrNotFound
+		}
+		return nil, "", err
+	}
+	return d.fs, objFile, nil
```

Comment:
Style: add an empty line between the close bracket of the if and the return clause.